engines/sci/graphics/menu.cpp

```diff
@@ -584,19 +596,35 @@ void GfxMenu::drawMenu(uint16 oldMenuId, uint16 newMenuId) {
 	_menuRect.top = _ports->_menuBarRect.bottom;
 	menuTextRect.top = _ports->_menuBarRect.top;
 	menuTextRect.bottom = _ports->_menuBarRect.bottom;
-	menuTextRect.left = menuTextRect.right = 7;
+	if (!g_sci->isLanguageRTL())
+		menuTextRect.left = menuTextRect.right = 7;
+	else
+		menuTextRect.left = menuTextRect.right = _ports->_menuBarRect.right - 7;
 	listIterator = _list.begin();
 	while (listIterator != listEnd) {
 		listEntry = *listIterator;
 		listNr++;
-		menuTextRect.left = menuTextRect.right;
-		menuTextRect.right += listEntry->textWidth;
-		if (listNr == newMenuId)
-			_menuRect.left = menuTextRect.left;
+		if (!g_sci->isLanguageRTL()) {
+			menuTextRect.left = menuTextRect.right;
+			menuTextRect.right += listEntry->textWidth;
+			if (listNr == newMenuId)
+				_menuRect.left = menuTextRect.left;
+		} else {
+			menuTextRect.right = menuTextRect.left;
+			menuTextRect.left -= listEntry->textWidth;
+			if (listNr == newMenuId)
+				_menuRect.right = menuTextRect.right;
+		}
 		if ((listNr == newMenuId) || (listNr == oldMenuId)) {
-			menuTextRect.translate(1, 0);
-			_paint16->invertRect(menuTextRect);
-			menuTextRect.translate(-1, 0);
+			if (!g_sci->isLanguageRTL()) {
+				menuTextRect.translate(1, 0);
+				_paint16->invertRect(menuTextRect);
+				menuTextRect.translate(-1, 0);
+			} else {
+				menuTextRect.translate(-1, 0);
+				_paint16->invertRect(menuTextRect);
+				menuTextRect.translate(1, 0);
+			}
```

Comment:
There's no need for a new block here. You can multiply with a value, e.g:

    int multiplier = !g_sci->isLanguageRTL() ? 1 : -1;
    ....
    menuTextRect.translate(1 * multiplier, 0);
    _paint16->invertRect(menuTextRect);
    menuTextRect.translate(-1 * multiplier, 0);